src/sage/combinat/words/morphism.py

```diff
@@ -1725,11 +1727,19 @@ def is_uniform(self, k=None):
             False
             sage: tau.is_uniform(k=2)
             True
+
+        TESTS:
+
+            sage: phi = WordMorphism('')
+            sage: phi.is_uniform()
+            False
         """
         if k is None:
-            return len(set(w.length() for w in self.images())) == 1
-        else:
-            return all(w.length() == k for w in self.images())
+            try:
+                k = self.images()[0].length()
+            except IndexError:
+                return False
```

Comment:
Do we consider these 4 lines an improvement over the 1 line that was there before?

Taking into account my comment above, I would just suggest:
```
        if k is None:
            return len(set(w.length() for w in self.images())) <= 1
```

Reply:
This avoids to compute all of them in case of failure (similar to `all` and `any`).

Reply:
Thanks, I missed that.